File: WebNovelApp/app/src/main/java/com/sinhvien/webnovelapp/models/BookmarkStatusResponse.kt
package com.sinhvien.webnovelapp.models

import com.google.gson.annotations.SerializedName

// Phản hồi khi kiểm tra trạng thái bookmark của một truyện
data class BookmarkStatusResponse(
    // Trạng thái: TRUE nếu đã bookmark, FALSE nếu chưa
    @SerializedName("IsBookmarked") val isBookmarked: Boolean,

    // ID của bookmark nếu nó tồn tại (giúp dễ dàng cho các thao tác tiếp theo)
    @SerializedName("BookmarkId") val bookmarkId: Int?,

    // Loại bookmark hiện tại
    @SerializedName("BookmarkType") val bookmarkType: String?
)

File: DarkNovel/DarkNovel/Models/ApiModels/BookmarkStatusResponse.cs
namespace DarkNovel.Models.ApiModels
{
    // Phải khớp với Data Class 'BookmarkStatusResponse' bên Kotlin
    public class BookmarkStatusResponse
    {
        // Phải khớp với @SerializedName("IsBookmarked")
        public bool IsBookmarked { get; set; }

        // Phải khớp với @SerializedName("BookmarkId")
        public int? BookmarkId { get; set; } // Dùng int? vì có thể null

        // Phải khớp với @SerializedName("BookmarkType")
        public string BookmarkType { get; set; }
    }
}